R/unnecessary_lambda_linter.R

```diff
@@ -4,6 +4,19 @@
 #'   e.g. `lapply(DF, sum)` is the same as `lapply(DF, function(x) sum(x))` and
 #'   the former is more readable.
 #'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "lapply(list(1:3, 2:4), function(xi) sum(xi))",
+#'   linters = unnecessary_lambda_linter()
+#' )
+#'
+#' # okay
```

Comment:
two more OK examples to add:

`lapply(x, \(xi) grep("ptn", xi)` [i.e., where the arg doesn't come first]
`lapply(x, \(xi) data.frame(col = xi))` [i.e., where the argument is named